### PR TITLE
Refactor Filter Tabs

### PR DESCRIPTION
**Pull request by michalsmiarowski**
Main changes:
- rename FilterTabs to FilterTabList and it's context
- remove `tabs` prop from `FilterTabList`
- refactor interface for FilterTab
- clean up and move components to separate directories

# Changes details

### rename FilterTabs to FilterTabList and it's context
FilterTabs -> FilterTabList
FilterTabsContext -> FilterTabListContext
TabsProviderProps -> FilterTabListProviderProps
FilterTabsProvider -> FilterTabListProvider

### remove `tabs` prop from `FilterTabList

Thanks to this change we can now write cleaner code. We can now use `FilterTab` component as a child of `FilterTabList` like this:

```
<FilterTabList {...props}>
	<FilterTab tabId={"specific-tab-id-1"}>Tab1</FilterTab>
	<FilterTab tabId={"specific-tab-id-2"}>Tab2</FilterTab>
</FilterTabList>
```

instead of:

```
<FilterTabs tabs={[
	{ title: "Tab1", tabId: "specific-tab-id-1" }
	{ title: "Tab2", tabId: "specific-tab-id-2" }
	]}
/>
```

This way we can add custom props to `FilterTab` (for example `as={Link}`)

### refactor interface for FilterTab

When using the FilterTab wrapped in another component (using `as` property) the properties of the wrapper component were not inherited. For example in such case:

```
import { Link as RouterLink } from "react-router-dom"

{...}

<FilterTab tabId={"specific-tab-id-1"} as={RouterLink} to={"/tab1"}>Tab1</FilterTab>
```

The `to` property was giving an error. To fix this I refactored the `FilterTab` interface so now it extends the `ChakraComponent`:

```
export interface FilterTabComponent
  extends ChakraComponent<"button", { tabId: string }> {}
```

### clean up and move components to separate directories

We had only one file for `FilterTabs` which was `FilterTabs/index.tsx`. It got little messy when we put context, provider, all interfaces, and both `FilterTabList` and `FilterTab` components inside it so I put all of it into separate directories:

```
.
├── FilterTabs
│   ├── context
│   │   ├── FilterTabListContext.tsx
│   │   └── index.tsx
│   ├── FilterTab.tsx
│   ├── FilterTabList.tsx
│   ├── FilterTabs.stories.tsx
│   └── index.tsx
└── ...
```

### CODE REVIEW

File: src/components/FilterTabs/FilterTab.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export interface FilterTabComponent
+  extends ChakraComponent<"button", { tabId: string }> {}
+
+export const FilterTab: FilterTabComponent = ({
+  tabId,
+  children,
+  ...props
+}) => {
+  const { selectedTabId, variant, onTabClick } =
+    useContext(FilterTabListContext)
```
Comment (r-czajkowski) [inline src/components/FilterTabs/FilterTab.tsx:14]: Can we create a wrapper for it in `/context/FilterTabListContext`? I mean:
```
const useFilterTablListContext = () => {
 return useContext(FilterTabListContext)
}
```

File: src/components/FilterTabs/FilterTabList.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { ButtonGroupProps, useStyleConfig, ButtonGroup } from "@chakra-ui/react"
+import { FC } from "react"
+import { FilterTabVariant, FilterTabSize } from "../../theme/FilterTab"
+import { FilterTabListProvider } from "./context"
+
+export interface FilterTabListProps extends ButtonGroupProps {
```
Comment (r-czajkowski) [inline src/components/FilterTabs/FilterTabList.tsx:6]: Maybe just `FilterTbas`? I think `List` is unnecessary. If you agree let's get rid of `List` in other places.

**Review by r-czajkowski — APPROVED** [review]
LGTM just left non-blocking commetns.

**Comment by georgeweiler** [comment]
LGTM 👍

**Review by georgeweiler — APPROVED** [review]

**Review by georgeweiler — CHANGES_REQUESTED** [review]
I think dark mode styles need updating.

see: https://www.figma.com/file/zZi2fYDUjWEMPQJWAt8VWv/Threshold-DS?node-id=2254%3A18276

**Comment by georgeweiler** [comment]
@michalsmiarowski I've pushed an update to the styling so it matches the design system. Hope thats okay. looks good to me. Happy to merge now 👍

**Review by georgeweiler — APPROVED** [review]

File: src/components/FilterTabs/FilterTab.tsx (hunk 1 above)
Comment (michalsmiarowski) [reply src/components/FilterTabs/FilterTab.tsx]: Fixed in c7ec175db8710f0213c8d3e9f35d8d2527371d93

File: src/components/FilterTabs/FilterTabList.tsx (hunk 2 above)
Comment (michalsmiarowski) [reply src/components/FilterTabs/FilterTabList.tsx]: Fixed in af7585f97f9f527a265a92441fd248f215044f86

File: src/theme/FilterTabs.ts [hunk 3]
```diff
@@ -8,7 +8,9 @@ export const FilterTabs: ComponentStyleConfig = {
   },
   variants: {
     primary: (props) => ({
-      bgColor: mode("white", "gray.900")(props),
+      bgColor: mode("white", "gray.800")(props),
+      border: "1px solid",
+      borderColor: "gray.100",
```
Comment (michalsmiarowski) [inline src/theme/FilterTabs.ts:13]: @georgeweiler  Based on Figma the border color of FilterTabs for "primary" vairant should be `E2E8F0` which is `gray.100`. I fixed that in c49a1d281e219cfb33dcf2f7394ba60290b10c09

**Review by georgeweiler — APPROVED** [review]

**Pull request merged**